ui/app/components/app/wallet-view.js

```diff
@@ -151,15 +151,23 @@ WalletView.prototype.render = function () {
   let type
   if (keyring) {
     type = keyring.type
-    if (type !== 'HD Key Tree') {
-      if (type.toLowerCase().search('hardware') !== -1) {
+    switch (type) {
+      case 'Trezor Hardware':
+      case 'Ledger Hardware':
         label = this.context.t('hardware')
-      } else {
+        break
+      case 'Simple Key Pair':
         label = this.context.t('imported')
-      }
+        break
+      case 'External Account':
+        label = this.context.t('external')
+        break
+      default:
+        label = ''
```

Comment:
maybe better to throw an error here

Reply:
Nope. The default keyring does not have a label, only the imported ones, the hardware, and bidirectional QR ones (external). So it is normal operation.